File: app/src/main/java/com/example/life/efficiency/views/MultiViewManager.java
package com.example.life.efficiency.views;

import android.view.View;

import java.util.Map;

public class MultiViewManager {

    private final Map<String, ActiveView> viewToLayoutId;

    public MultiViewManager(Map<String, ActiveView> viewToLayoutId) {
        this.viewToLayoutId = viewToLayoutId;
        setupButtons();
    }

    public void setupButtons() {
        for (final String viewName : viewToLayoutId.keySet()) {
            ActiveView thisView = viewToLayoutId.get(viewName);
            if (thisView == null)
                continue;
            thisView.getButton().setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    try {
                        changeView(viewName);
                    } catch (ViewNotFoundException e) {
                        e.printStackTrace();
                    }
                }
            });
        }
    }

    public void changeView(String viewName) throws ViewNotFoundException {
        if (!viewToLayoutId.containsKey(viewName)) {
            throw new ViewNotFoundException(viewName);
        }

        for (ActiveView activeView : viewToLayoutId.values()) {
            if (activeView != viewToLayoutId.get(viewName)) activeView.getView().setVisibility(View.GONE);
            else {
                activeView.onActive();
                activeView.getView().setVisibility(View.VISIBLE);
            }
        }
    }

}

File: app/src/main/kotlin/com/leon/patmore/life/efficiency/views/MultiViewManager.kt
package com.leon.patmore.life.efficiency.views

import android.view.View
import android.widget.Button

class MultiViewManager(private val menuView: View,
                       private val menuButton: Button,
                       private val viewToLayoutId: Map<String, ActiveView>) {

    init {
        setupButtons()
    }

    private fun setupButtons() {
        for (viewName in viewToLayoutId.keys) {
            val thisView = viewToLayoutId[viewName] ?: continue
            thisView.button.setOnClickListener { changeView(viewName) }
        }
        menuButton.setOnClickListener {
            for (activeView in viewToLayoutId.values) {
                activeView.view.visibility = View.GONE
            }
            menuView.visibility = View.VISIBLE
        }
    }

    private fun changeView(viewName: String) {
        if (!viewToLayoutId.containsKey(viewName)) {
            return
        }
        for (activeView in viewToLayoutId.values) {
            if (activeView !== viewToLayoutId[viewName]) activeView.view.visibility = View.GONE else {
                activeView.onActive()
                activeView.view.visibility = View.VISIBLE
            }
        }
        menuView.visibility = View.GONE
    }

}
